<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Haider Inam" />

<meta name="date" content="2024-12-23" />

<title>ErrorRates_Tileseq_NGS_Duplex_SSCS</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">abl_dms</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ErrorRates_Tileseq_NGS_Duplex_SSCS</h1>
<h4 class="author">Haider Inam</h4>
<h4 class="date">2024-12-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>abl_dms/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250621code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250621)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250621code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250621)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcompritchardlabatpsuabldmstreee047290ce0aa76112aadf86daf70a0f74eeb1551targetblanke047290a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/pritchardlabatpsu/abl_dms/tree/e047290ce0aa76112aadf86daf70a0f74eeb1551" target="_blank">e047290</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcompritchardlabatpsuabldmstreee047290ce0aa76112aadf86daf70a0f74eeb1551targetblanke047290a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/pritchardlabatpsu/abl_dms/tree/e047290ce0aa76112aadf86daf70a0f74eeb1551" target="_blank">e047290</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/Twinstrand/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/Enrichmentsimulations/.DS_Store
    Ignored:    output/spikeins_figures/.DS_Store

Untracked files:
    Untracked:  analysis/ErrorRates_NNMD&ROC_region1.Rmd
    Untracked:  analysis/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd
    Untracked:  analysis/ErrorRates_dataparser_region1.Rmd
    Untracked:  analysis/Homepage_errorrates.Rmd
    Untracked:  code/compare_samples.R
    Untracked:  code/compare_screens.R
    Untracked:  code/compare_screens_archive.R
    Untracked:  code/compare_screens_twosamples.R
    Untracked:  code/cosmic_data_adder.R
    Untracked:  code/depth_finder.R
    Untracked:  code/is_intended_adder.R
    Untracked:  code/merge_samples.R
    Untracked:  code/merge_samples_twosamples.R
    Untracked:  code/plotting/
    Untracked:  code/res_residues_adder.R
    Untracked:  code/resmuts_adder.R
    Untracked:  code/shortest_codon_finder.R
    Untracked:  code/variants_parser.R
    Untracked:  data/ABL1AppOutput/
    Untracked:  data/Consensus_Data/
    Untracked:  data/Refs/
    Untracked:  data/codon_table.csv
    Untracked:  data/twinstrand_maf_merge_formanuscript.csv
    Untracked:  data/twinstrand_simple_melt_merge copy.csv
    Untracked:  output/ABLEnrichmentScreens/
    Untracked:  output/errorrates_figures/
    Untracked:  output/spikeins_figures/dosing_normalization_stdevs_paired.pdf

Unstaged changes:
    Modified:   analysis/index.Rmd
    Modified:   analysis/spikeins_growthrates.Rmd
    Deleted:    dosing_normalization_stdevs_paired.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>source(&quot;code/resmuts_adder.R&quot;)
#Cleanup code for plotting
cleanup=theme_bw() +
  theme(plot.title = element_text(hjust=.5),
        panel.grid.major = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.background = element_blank(),
        # axis.line = element_line(color = &quot;black&quot;),
        axis.text = element_text(face=&quot;bold&quot;,color=&quot;black&quot;,size=&quot;11&quot;),
        text=element_text(size=11,face=&quot;bold&quot;),
        axis.title=element_text(face=&quot;bold&quot;,size=&quot;11&quot;))</code></pre>
<p>Doing an NGS, SSCS, DCS, and Tileseq comparison</p>
<p>The following data was taken from Marta This is lane 18b sample 8 vs
10 duplex and tileseq data</p>
<pre class="r"><code>#Cleanup code for plotting
source(&quot;code/compare_screens_archive.R&quot;)

source(&quot;code/plotting/cleanup.R&quot;)
source(&quot;code/cosmic_data_adder.R&quot;)
# source(&quot;code/compare_screens.R&quot;)

source(&quot;code/compare_screens_twosamples.R&quot;)
source(&quot;code/merge_samples_twosamples.R&quot;)
source(&quot;code/plotting/heatmap_plotting_function.R&quot;)</code></pre>
<div id="data-plotting." class="section level2">
<h2>Data Plotting.</h2>
<p>This sequencing data was generated using the scripts labeled
ErrorRates_dataparser_region1.</p>
<pre class="r"><code>data=read.csv(&quot;output/ABLEnrichmentScreens/ABL_Region1_lane18b/baf3_Imat_high_rep1vsrep2_duplex/screen_comparison_baf3_Imat_high_rep1vsrep2_duplex.csv&quot;,header = T,stringsAsFactors = F)[-1]
data$seqtech=&quot;duplex&quot;
data_duplex=data


data=read.csv(&quot;output/ABLEnrichmentScreens/ABL_Region1_lane18b/baf3_Imat_high_rep1vsrep2_ngs/screen_comparison_baf3_Imat_high_rep1vsrep2_ngs.csv&quot;,header = T,stringsAsFactors = F)[-1]
data$seqtech=&quot;ngs&quot;
data_ngs=data

data=read.csv(&quot;output/ABLEnrichmentScreens/ABL_Region1_lane18b/baf3_Imat_high_rep1vsrep2_sscs/screen_comparison_baf3_Imat_high_rep1vsrep2_sscs.csv&quot;,header = T,stringsAsFactors = F)[-1]
data$seqtech=&quot;sscs&quot;
data_sscs=data

data=read.csv(&quot;output/ABLEnrichmentScreens/ABL_Region1_lane18b/baf3_Imat_high_rep1vsrep2_tileseq/screen_comparison_baf3_Imat_high_rep1vsrep2_tileseq.csv&quot;,header = T,stringsAsFactors = F)[-1]
data$seqtech=&quot;tileseq&quot;
data_tileseq=data

data=rbind(data_duplex,data_ngs,data_sscs,data_tileseq)

data_sum=data%&gt;%
  filter(protein_start%in%c(242:322),is_intended%in%1)%&gt;%
  group_by(seqtech)%&gt;%
  summarize(mean_depth=mean(depth_screen1_before),
            num=n())

data=data%&gt;%rowwise%&gt;%mutate(n_nuc=nchar(alt))</code></pre>
</div>
<div id="data-plotting" class="section level1">
<h1>Data plotting</h1>
<div
id="plotting-unintended-vs-intended-mutant-distributions-across-sequencing-technologies."
class="section level3">
<h3>Plotting unintended vs intended mutant distributions across
sequencing technologies.</h3>
<pre class="r"><code>ggplot(data%&gt;%filter(ct_screen1_before&gt;=5,seqtech%in%&quot;ngs&quot;,n_nuc%in%1),aes(x=ct_screen1_before,fill=factor(is_intended,levels=c(1,0))))+geom_density(alpha=0.7)+scale_x_continuous(&quot;Number of times a variant was seen&quot;,trans=&quot;log10&quot;)+cleanup+scale_fill_manual(values = c(&quot;blue&quot;,&quot;red&quot;))+scale_y_continuous(&quot;Density&quot;)+theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;output/errorrates_figures/plot6a_abl_ngs_untreated.pdf&quot;,width=3.5,height=2.5,units=&quot;in&quot;,useDingbats=F)

ggplot(data%&gt;%filter(ct_screen1_before&gt;=5,seqtech%in%&quot;tileseq&quot;,n_nuc%in%1),aes(x=ct_screen1_before,fill=factor(is_intended,levels=c(1,0))))+geom_density(alpha=0.7)+scale_x_continuous(&quot;Number of times a variant was seen&quot;,trans=&quot;log10&quot;)+cleanup+scale_fill_manual(values = c(&quot;blue&quot;,&quot;red&quot;))+scale_y_continuous(&quot;Density&quot;)+theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;output/errorrates_figures/plot6b_abl_tileseq_untreated.pdf&quot;,width=3.5,height=2.5,units=&quot;in&quot;,useDingbats=F)

ggplot(data%&gt;%filter(seqtech%in%&quot;duplex&quot;,n_nuc%in%1),aes(x=ct_screen1_before,fill=factor(is_intended,levels=c(1,0))))+geom_density(alpha=0.7)+scale_x_continuous(&quot;Number of times a variant was seen&quot;,trans=&quot;log10&quot;)+cleanup+scale_fill_manual(values = c(&quot;blue&quot;,&quot;red&quot;))+scale_y_continuous(&quot;Density&quot;)+theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;output/errorrates_figures/plot6c_abl_duplex_untreated.pdf&quot;,width=3.5,height=2.5,units=&quot;in&quot;,useDingbats=F)</code></pre>
</div>
<div id="plots" class="section level2">
<h2>Plots:</h2>
<div id="signal-to-noise-ratios" class="section level3">
<h3>Signal to noise ratios</h3>
</div>
<div id="unintended-mutants-coverage" class="section level3">
<h3>Unintended mutants coverage</h3>
</div>
<div id="coverage-of-intended-resistant-mutants" class="section level3">
<h3>Coverage of intended resistant mutants</h3>
</div>
<div id="cdf-across-sequencing-technologies" class="section level3">
<h3>CDF across sequencing technologies</h3>
</div>
</div>
<div id="plots-for-figures-in-paper." class="section level2">
<h2>Plots for figures in paper.</h2>
<pre class="r"><code>library(tidyr)
data_snr=data%&gt;%filter(protein_start%in%c(242:322))%&gt;%group_by(seqtech,is_intended,protein_start,n_nuc_min)%&gt;%summarize(nmuts=n(),coverage_mean=mean(ct_screen1_before),coverage_sum=sum(ct_screen1_before),depth_mean=mean(depth_screen1_before))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;seqtech&#39;, &#39;is_intended&#39;, &#39;protein_start&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>data_snr_cast &lt;- data_snr %&gt;%
  pivot_wider(
    names_from = is_intended,
    values_from = c(nmuts,coverage_mean,coverage_sum,depth_mean),
    names_glue = &quot;{.value}_{ifelse(is_intended == 1, &#39;intended&#39;, &#39;unintended&#39;)}&quot;
  )
# data_snr_cast=data_snr_cast%&gt;%mutate(snr=coverage_mean_intended/coverage_mean_unintended)
data_snr_cast=data_snr_cast%&gt;%mutate(snr=coverage_sum_intended/coverage_sum_unintended)

# Demonstrating noise:
ggplot(data_snr_cast%&gt;%filter(seqtech%in%c(&quot;duplex&quot;,&quot;tileseq&quot;)),aes(x=protein_start,y=coverage_sum_unintended))+
  geom_point(color=&quot;black&quot;,size=2,shape=21,aes(fill=seqtech))+
  # facet_wrap(~seqtech)+
  scale_fill_manual(values = c(&quot;#FB8072&quot;,&quot;#8ED3C7&quot;))+
  scale_y_continuous(&quot;Coverage of unintended mutants&quot;,trans = &quot;log10&quot;)+
  scale_x_continuous(&quot;Rediue on BCRABL&quot;)+
  cleanup+
  theme(legend.position = &quot;none&quot;,
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color=&quot;black&quot;,size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))</code></pre>
<pre><code>Warning: Removed 240 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;output/errorrates_figures/plot1.pdf&quot;,width=2.5,height=2.5,units = &quot;in&quot;,useDingbats=F)</code></pre>
<pre><code>Warning: Removed 240 rows containing missing values (geom_point).</code></pre>
<pre class="r"><code>ggplot(data_snr_cast%&gt;%filter(seqtech%in%c(&quot;duplex&quot;,&quot;tileseq&quot;)),aes(x=protein_start,y=nmuts_unintended,fill=seqtech))+
  geom_point(color=&quot;black&quot;,size=2,shape=21,aes(fill=seqtech))+
  scale_fill_manual(values = c(&quot;#FB8072&quot;,&quot;#8ED3C7&quot;))+
  geom_hline(yintercept = 40,linetype=&quot;dashed&quot;)+
  cleanup+
  theme(legend.position = &quot;none&quot;,
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color=&quot;black&quot;,size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))</code></pre>
<pre><code>Warning: Removed 240 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;output/errorrates_figures/plot2.pdf&quot;,width=2.5,height=2.5,units = &quot;in&quot;,useDingbats=F)</code></pre>
<pre><code>Warning: Removed 240 rows containing missing values (geom_point).</code></pre>
<pre class="r"><code># Demonstrating coverage:
ggplot(data%&gt;%filter(resmuts%in%T,is_intended%in%1,seqtech%in%c(&quot;duplex&quot;,&quot;tileseq&quot;)),aes(x=species,y=ct_screen1_before,fill=seqtech))+
  geom_point(color=&quot;black&quot;,shape=21)+
  geom_line()+
  scale_y_continuous(&quot;Day 0 Coverage of Resistant Mutant&quot;,trans=&quot;log10&quot;,limits=c(1,1.2e5))+
  cleanup+
  theme(axis.text.x=element_text(color=&quot;black&quot;,size=8,angle=90,hjust=.5,vjust=.5),
        legend.position = &quot;none&quot;,
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))</code></pre>
<pre><code>geom_path: Each group consists of only one observation. Do you need to adjust
the group aesthetic?</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;output/errorrates_figures/plot3.pdf&quot;,width=2.5,height=2.5,units = &quot;in&quot;,useDingbats=F)</code></pre>
<pre><code>geom_path: Each group consists of only one observation. Do you need to adjust
the group aesthetic?</code></pre>
<pre class="r"><code>ggplot(data%&gt;%filter(resmuts%in%T,is_intended%in%1,seqtech%in%c(&quot;duplex&quot;,&quot;tileseq&quot;)),aes(x=species,y=ct_screen2_after,fill=seqtech))+
  geom_point(color=&quot;black&quot;,shape=21)+
  geom_line()+
  scale_y_continuous(&quot;Day 6 Coverage of Resistant Mutant&quot;,trans=&quot;log10&quot;,limits=c(1,1.2e5))+
  cleanup+
  theme(axis.text.x=element_text(color=&quot;black&quot;,size=8,angle=90,hjust=.5,vjust=.5),
        legend.position = &quot;none&quot;,
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))</code></pre>
<pre><code>geom_path: Each group consists of only one observation. Do you need to adjust
the group aesthetic?</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;output/errorrates_figures/plot3b.pdf&quot;,width=2.5,height=2.5,units = &quot;in&quot;,useDingbats=F)</code></pre>
<pre><code>geom_path: Each group consists of only one observation. Do you need to adjust
the group aesthetic?</code></pre>
<pre class="r"><code>ggplot(data%&gt;%filter(is_intended%in%1,seqtech%in%c(&quot;duplex&quot;,&quot;tileseq&quot;)),aes(x=ct_screen1_before,color=seqtech))+
  stat_ecdf()+
  scale_x_continuous(&quot;Intended mutant coverage&quot;,trans=&quot;log10&quot;)+
  scale_y_continuous(&quot;CDF&quot;)+
  scale_color_manual(values=c(&quot;#FB8072&quot;,&quot;#8ED3C7&quot;))+
  cleanup+
  theme(axis.text.x=element_text(color=&quot;black&quot;,size=8,angle=90,hjust=.5,vjust=.5),
        legend.position = &quot;none&quot;,
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-4-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;output/errorrates_figures/plot4.pdf&quot;,width=2.5,height=2.5,units = &quot;in&quot;,useDingbats=F)</code></pre>
<pre class="r"><code>ggplot(data_snr_cast%&gt;%filter(seqtech%in%c(&quot;duplex&quot;,&quot;tileseq&quot;,&quot;ngs&quot;)),aes(x=protein_start,y=coverage_sum_unintended))+
  geom_point(color=&quot;black&quot;,size=2,shape=21,aes(fill=seqtech))+
  # facet_wrap(~seqtech)+
  scale_fill_manual(values = c(&quot;#FB8072&quot;,&quot;#3287BD&quot;,&quot;#8ED3C7&quot;))+
  scale_y_continuous(&quot;Coverage of unintended mutants&quot;,trans = &quot;log10&quot;)+
  scale_x_continuous(&quot;Rediue on BCRABL&quot;)+
  cleanup+
  theme(legend.position = &quot;none&quot;,
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color=&quot;black&quot;,size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))</code></pre>
<pre><code>Warning: Removed 277 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data_snr_cast%&gt;%filter(n_nuc_min%in%1,seqtech%in%c(&quot;duplex&quot;,&quot;tileseq&quot;,&quot;ngs&quot;)),aes(x=protein_start,y=nmuts_unintended,fill=seqtech))+
  geom_point(color=&quot;black&quot;,size=2,shape=21,aes(fill=seqtech))+
  geom_line(aes(color=seqtech))+
  scale_fill_manual(values = c(&quot;#FB8072&quot;,&quot;#3287BD&quot;,&quot;#8ED3C7&quot;))+
  scale_color_manual(values = c(&quot;#FB8072&quot;,&quot;#3287BD&quot;,&quot;#8ED3C7&quot;))+
  # geom_hline(yintercept = 40,linetype=&quot;dashed&quot;)+
  cleanup+
  theme(legend.position = &quot;none&quot;,
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color=&quot;black&quot;,size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))</code></pre>
<pre><code>Warning: Removed 52 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 1 row(s) containing missing values (geom_path).</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="plotting-mutant-standards." class="section level2">
<h2>Plotting mutant standards.</h2>
<p>Looking at the imatinib resistant of known clinically imatR
mutants</p>
<pre class="r"><code>library(binom)

standards=read.csv(&quot;data/ic50data/ic50data_all_conc.csv&quot;,header = T,stringsAsFactors = F)
standards=standards%&gt;%filter(conc%in%1.2)

# data=data%&gt;%filter(is_intended%in%1)
################### Standards with NGS data###############
standards_ngs=merge(data%&gt;%filter(seqtech%in%&quot;ngs&quot;,is_intended%in%1),standards,by.x=&quot;species&quot;,by.y=&quot;mutant&quot;)

standards_ngs=standards_ngs%&gt;%
  rowwise()%&gt;%
  mutate(
    mean_fitness = mean(c(netgr_obs_screen1, netgr_obs_screen2)),
    sd_fitness = sd(c(netgr_obs_screen1, netgr_obs_screen2)),
    se_fitness = sd_fitness / sqrt(2),  # SE with two replicates
    lower_ci = mean_fitness - 1.96 * se_fitness,  # 95% CI lower bound
    upper_ci = mean_fitness + 1.96 * se_fitness   # 95% CI upper bound
  )
################### Standards with tileseq data###############
standards_tileseq=merge(data%&gt;%filter(seqtech%in%&quot;tileseq&quot;,is_intended%in%1),standards,by.x=&quot;species&quot;,by.y=&quot;mutant&quot;)

standards_tileseq=standards_tileseq%&gt;%
  rowwise()%&gt;%
  mutate(
    mean_fitness = mean(c(netgr_obs_screen1, netgr_obs_screen2)),
    sd_fitness = sd(c(netgr_obs_screen1, netgr_obs_screen2)),
    se_fitness = sd_fitness / sqrt(2),  # SE with two replicates
    lower_ci = mean_fitness - 1.96 * se_fitness,  # 95% CI lower bound
    upper_ci = mean_fitness + 1.96 * se_fitness   # 95% CI upper bound
  )
################### Standards with dupseq data###############
standards_duplex=merge(data%&gt;%filter(seqtech%in%&quot;duplex&quot;,is_intended%in%1),standards,by.x=&quot;species&quot;,by.y=&quot;mutant&quot;)

standards_duplex=standards_duplex%&gt;%
  rowwise()%&gt;%
  mutate(
    mean_fitness = mean(c(netgr_obs_screen1, netgr_obs_screen2)),
    sd_fitness = sd(c(netgr_obs_screen1, netgr_obs_screen2)),
    se_fitness = sd_fitness / sqrt(2),  # SE with two replicates
    lower_ci = mean_fitness - 1.96 * se_fitness,  # 95% CI lower bound
    upper_ci = mean_fitness + 1.96 * se_fitness   # 95% CI upper bound
  )
standards_duplex$netgr_pred=0.055-standards_duplex$drug_effect

#########################################################


ggplot()+
  geom_point(size=2,shape=21,fill=&quot;#3287BD&quot;,aes(x=.055-standards_ngs$drug_effect,y=standards_ngs$mean_fitness))+
  geom_errorbar(aes(x=.055-standards_ngs$drug_effect,ymin=standards_ngs$lower_ci,ymax=standards_ngs$upper_ci,width=0.001))+
  stat_cor(aes(x=.055-standards_ngs$drug_effect,y=standards_ngs$mean_fitness),method=&quot;pearson&quot;,color=&quot;#3287BD&quot;,size=2,vjust=3)+
  geom_point(size=2,shape=21,fill=&quot;#8ED3C7&quot;,aes(x=.055-standards_tileseq$drug_effect,y=standards_tileseq$mean_fitness))+
  geom_errorbar(aes(x=.055-standards_tileseq$drug_effect,ymin=standards_tileseq$lower_ci,ymax=standards_tileseq$upper_ci,width=0.001))+
  stat_cor(aes(x=.055-standards_tileseq$drug_effect,y=standards_tileseq$mean_fitness),method=&quot;pearson&quot;,color=&quot;#8ED3C7&quot;,size=2,vjust=1.5)+
  geom_point(size=2,shape=21,fill=&quot;#FB8072&quot;,aes(x=.055-standards_duplex$drug_effect,y=standards_duplex$mean_fitness))+
  geom_errorbar(aes(x=.055-standards_duplex$drug_effect,ymin=standards_duplex$lower_ci,ymax=standards_duplex$upper_ci,width=0.001))+
  stat_cor(aes(x=.055-standards_duplex$drug_effect,y=standards_duplex$mean_fitness),method=&quot;pearson&quot;,color=&quot;#FB8072&quot;,size=2,vjust=0)+
  
  
  scale_x_continuous(&quot;Predicted Net Growth Rate (Hrs-1) \nClonal IC50&quot;,limits=c(-0.016,.06))+
  scale_y_continuous(&quot;Observed Net Growth Rate (Hrs-1) \nDMS Screen&quot;,limits=c(-0.016,.06))+
  geom_abline(linetype=&quot;dashed&quot;)+
  cleanup+
  theme(legend.position = c(0.92,0.45),
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color=&quot;black&quot;,size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))

ggsave(&quot;output/errorrates_figures/plot5.pdf&quot;,width=2.5,height=2.5,units = &quot;in&quot;,useDingbats=F)

ggplot()+
  geom_point(size=2,shape=21,fill=&quot;#3287BD&quot;,aes(x=.055-standards_ngs$drug_effect,y=standards_ngs$mean_fitness))+
  geom_errorbar(aes(x=.055-standards_ngs$drug_effect,ymin=standards_ngs$lower_ci,ymax=standards_ngs$upper_ci,width=0.001))+
  stat_cor(aes(x=.055-standards_ngs$drug_effect,y=standards_ngs$mean_fitness),method=&quot;pearson&quot;,color=&quot;#3287BD&quot;,size=2,vjust=3)+
scale_x_continuous(&quot;Predicted Net Growth Rate (Hrs-1) \nClonal IC50&quot;,limits=c(-0.016,.06))+
  scale_y_continuous(&quot;Observed Net Growth Rate (Hrs-1) \nDMS Screen&quot;,limits=c(-0.016,.06))+
  geom_abline(linetype=&quot;dashed&quot;)+
  cleanup+
  theme(legend.position = c(0.92,0.45),
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color=&quot;black&quot;,size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))
ggsave(&quot;output/errorrates_figures/plot5_ngs.pdf&quot;,width=2.5,height=2.5,units = &quot;in&quot;,useDingbats=F)

ggplot()+
geom_point(size=2,shape=21,fill=&quot;#8ED3C7&quot;,aes(x=.055-standards_tileseq$drug_effect,y=standards_tileseq$mean_fitness))+
  geom_errorbar(aes(x=.055-standards_tileseq$drug_effect,ymin=standards_tileseq$lower_ci,ymax=standards_tileseq$upper_ci,width=0.001))+
  stat_cor(aes(x=.055-standards_tileseq$drug_effect,y=standards_tileseq$mean_fitness),method=&quot;pearson&quot;,color=&quot;#8ED3C7&quot;,size=2,vjust=1.5)+
  scale_x_continuous(&quot;Predicted Net Growth Rate (Hrs-1) \nClonal IC50&quot;,limits=c(-0.016,.06))+
  scale_y_continuous(&quot;Observed Net Growth Rate (Hrs-1) \nDMS Screen&quot;,limits=c(-0.016,.06))+
  geom_abline(linetype=&quot;dashed&quot;)+
  cleanup+
  theme(legend.position = c(0.92,0.45),
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color=&quot;black&quot;,size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))
# ggsave(&quot;output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot5_tileseq.pdf&quot;,width=2.5,height=2.5,units = &quot;in&quot;,useDingbats=F)

ggplot()+
geom_point(size=2,shape=21,fill=&quot;#FB8072&quot;,aes(x=.055-standards_duplex$drug_effect,y=standards_duplex$mean_fitness))+
  geom_errorbar(aes(x=.055-standards_duplex$drug_effect,ymin=standards_duplex$lower_ci,ymax=standards_duplex$upper_ci,width=0.001))+
  stat_cor(aes(x=.055-standards_duplex$drug_effect,y=standards_duplex$mean_fitness),method=&quot;pearson&quot;,color=&quot;#FB8072&quot;,size=2,vjust=0)+
scale_x_continuous(&quot;Predicted Net Growth Rate (Hrs-1) \nClonal IC50&quot;,limits=c(-0.016,.06))+
  scale_y_continuous(&quot;Observed Net Growth Rate (Hrs-1) \nDMS Screen&quot;,limits=c(-0.016,.06))+
  geom_abline(linetype=&quot;dashed&quot;)+
  cleanup+
  theme(legend.position = c(0.92,0.45),
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color=&quot;black&quot;,size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))
ggsave(&quot;output/errorrates_figures/plot5_duplex.pdf&quot;,width=2.5,height=2.5,units = &quot;in&quot;,useDingbats=F)</code></pre>
</div>
<div id="plotting-mutant-standards-binomial-confidence-intervals"
class="section level2">
<h2>Plotting mutant standards binomial confidence intervals</h2>
<pre class="r"><code>library(binom)</code></pre>
<pre><code>Warning: package &#39;binom&#39; was built under R version 4.0.2</code></pre>
<pre class="r"><code>standards=read.csv(&quot;data/ic50data/ic50data_all_conc.csv&quot;,header = T,stringsAsFactors = F)
standards=standards%&gt;%filter(conc%in%1.2)

# data=data%&gt;%filter(is_intended%in%1)
################### Standards with NGS data###############
standards_ngs=merge(data%&gt;%filter(seqtech%in%&quot;ngs&quot;,is_intended%in%1),standards,by.x=&quot;species&quot;,by.y=&quot;mutant&quot;)

standards_ngs=standards_ngs%&gt;%
  rowwise()%&gt;%
  mutate(
    mean_fitness = mean(c(netgr_obs_screen1, netgr_obs_screen2)),
    sd_fitness = sd(c(netgr_obs_screen1, netgr_obs_screen2)),
    se_fitness = sd_fitness / sqrt(2),  # SE with two replicates
    lower_ci = mean_fitness - 1.96 * se_fitness,  # 95% CI lower bound
    upper_ci = mean_fitness + 1.96 * se_fitness   # 95% CI upper bound
  )
################### Standards with tileseq data###############
standards_tileseq=merge(data%&gt;%filter(seqtech%in%&quot;tileseq&quot;,is_intended%in%1),standards,by.x=&quot;species&quot;,by.y=&quot;mutant&quot;)

standards_tileseq=standards_tileseq%&gt;%
  rowwise()%&gt;%
  mutate(
    mean_fitness = mean(c(netgr_obs_screen1, netgr_obs_screen2)),
    sd_fitness = sd(c(netgr_obs_screen1, netgr_obs_screen2)),
    se_fitness = sd_fitness / sqrt(2),  # SE with two replicates
    lower_ci = mean_fitness - 1.96 * se_fitness,  # 95% CI lower bound
    upper_ci = mean_fitness + 1.96 * se_fitness   # 95% CI upper bound
  )
################### Standards with dupseq data###############
standards_duplex=merge(data%&gt;%filter(seqtech%in%&quot;duplex&quot;,is_intended%in%1),standards,by.x=&quot;species&quot;,by.y=&quot;mutant&quot;)

standards_duplex=standards_duplex%&gt;%
  rowwise()%&gt;%
  mutate(
    mean_fitness = mean(c(netgr_obs_screen1, netgr_obs_screen2)),
    sd_fitness = sd(c(netgr_obs_screen1, netgr_obs_screen2)),
    se_fitness = sd_fitness / sqrt(2),  # SE with two replicates
    lower_ci = mean_fitness - 1.96 * se_fitness,  # 95% CI lower bound
    upper_ci = mean_fitness + 1.96 * se_fitness   # 95% CI upper bound
  )
standards_duplex$netgr_pred=0.055-standards_duplex$drug_effect

#########################################################
# Counts for my reference
# Day 0: 89
# No drug rep 1: 123767
# No drug rep 2: 94619
# High drug rep 1: 5775
# High drug rep 2: 6240
# this function takes count, depth, and ct_before, and ct_after counts. Then, this function returns the binomial confidence intervals around the ct_after counts
 # 89 initiated as teh before count because that was the cells/mL count on D0 in our high imatinib dose condition for R1
  # 5775 initiated as teh before count because that was the cells/mL count on D0 in our high imatinib dose condition for R1
binom_netgr=function(ct_before,
                     depth_before,
                     ct_after,
                     depth_after,
                     deltat=144,
                     cells_before=89,
                     cells_after){
  ct_before=round(ct_before,0)
  depth_before=round(depth_before,0)
  ct_after=round(ct_after,0)
  depth_after=round(depth_after,0)
  # ct_before=2
  # depth_before=6745
  # ct_after=3
  # depth_after=4665
  # deltat=144
  # cells_before=89
  # cells_after=5775
  ct_before_ll=round(binom.test(ct_before,depth_before,conf.level = 0.95)$conf.int[1]*depth_before,0)
  ct_before_ul=round(binom.test(ct_before,depth_before,conf.level = 0.95)$conf.int[2]*depth_before,0)
  
  ct_after_ll=round(binom.test(ct_after,depth_after,conf.level = 0.95)$conf.int[1]*depth_after,0)
  ct_after_ul=round(binom.test(ct_after,depth_after,conf.level = 0.95)$conf.int[2]*depth_after,0)
  
  # Important: note how this estimate of the confidence interval of the net growth rate is using the confidence intervals from both the timepoints, not just one of the timepoints. Another option is to calculate confidence intervals using the count data from one of the timepoints..
  netgr_ll=round(log(((ct_after_ll/depth_after)*cells_after)/((ct_before_ul/depth_before)*cells_before))/deltat,4)
  netgr_ul=round(log(((ct_after_ul/depth_after)*cells_after)/((ct_before_ll/depth_before)*cells_before))/deltat,4)
  # Other way of doing this:
  # netgr_ll=log(((ct_after_ll/depth_after)*cells_after)/((ct_before/depth_before)*cells_before))/deltat
  list(netgr_ll,netgr_ul)
}
# binom_netgr(2,6000,6,4000,cells_after=5775)[1]

standards_duplex=standards_duplex%&gt;%
  rowwise()%&gt;%
  mutate(netgr_ll=unlist(binom_netgr(ct_screen1_before,
                              depth_screen1_before,
                              ct_screen1_after,
                              depth_screen1_after,
                              cells_after = 5775)[1]),
         netgr_ul=unlist(binom_netgr(ct_screen1_before,
                              depth_screen1_before,
                              ct_screen1_after,
                              depth_screen1_after,
                              cells_after = 5775)[2]))

standards_tileseq=standards_tileseq%&gt;%
  rowwise()%&gt;%
  mutate(netgr_ll=unlist(binom_netgr(ct_screen1_before,
                              depth_screen1_before,
                              ct_screen1_after,
                              depth_screen1_after,
                              cells_after = 5775)[1]),
         netgr_ul=unlist(binom_netgr(ct_screen1_before,
                              depth_screen1_before,
                              ct_screen1_after,
                              depth_screen1_after,
                              cells_after = 5775)[2]))

standards_ngs=standards_ngs%&gt;%
  rowwise()%&gt;%
  mutate(netgr_ll=unlist(binom_netgr(ct_screen1_before,
                              depth_screen1_before,
                              ct_screen1_after,
                              depth_screen1_after,
                              cells_after = 5775)[1]),
         netgr_ul=unlist(binom_netgr(ct_screen1_before,
                              depth_screen1_before,
                              ct_screen1_after,
                              depth_screen1_after,
                              cells_after = 5775)[2]))

ggplot()+
geom_point(size=2,shape=21,fill=&quot;#FB8072&quot;,aes(x=.055-standards_duplex$drug_effect,y=standards_duplex$netgr_obs_screen1))+
  geom_errorbar(aes(x=.055-standards_duplex$drug_effect,ymin=standards_duplex$netgr_ll,ymax=standards_duplex$netgr_ul,width=0.001))+
  stat_cor(aes(x=.055-standards_duplex$drug_effect,y=standards_duplex$mean_fitness),method=&quot;pearson&quot;,color=&quot;#FB8072&quot;,size=2,vjust=0)+
scale_x_continuous(&quot;Predicted Net Growth Rate (Hrs-1) \nClonal IC50&quot;,limits=c(-0.016,.06))+
  scale_y_continuous(&quot;Observed Net Growth Rate (Hrs-1) \nDMS Screen&quot;,limits=c(-0.016,.06))+
  geom_abline(linetype=&quot;dashed&quot;)+
  cleanup+
  theme(legend.position = c(0.92,0.45),
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color=&quot;black&quot;,size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;output/errorrates_figures/plot5_duplex.pdf&quot;,width=2.5,height=2.5,units = &quot;in&quot;,useDingbats=F)</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre class="r"><code>ggplot()+
geom_point(size=2,shape=21,fill=&quot;#8ED3C7&quot;,aes(x=.055-standards_tileseq$drug_effect,y=standards_tileseq$netgr_obs_screen1))+
  geom_errorbar(aes(x=.055-standards_tileseq$drug_effect,ymin=standards_tileseq$netgr_ll,ymax=standards_tileseq$netgr_ul,width=0.001))+
  stat_cor(aes(x=.055-standards_tileseq$drug_effect,y=standards_tileseq$mean_fitness),method=&quot;pearson&quot;,color=&quot;#8ED3C7&quot;,size=2,vjust=1.5)+
  scale_x_continuous(&quot;Predicted Net Growth Rate (Hrs-1) \nClonal IC50&quot;,limits=c(-0.016,.06))+
  scale_y_continuous(&quot;Observed Net Growth Rate (Hrs-1) \nDMS Screen&quot;,limits=c(-0.016,.06))+
  geom_abline(linetype=&quot;dashed&quot;)+
  cleanup+
  theme(legend.position = c(0.92,0.45),
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color=&quot;black&quot;,size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;output/errorrates_figures/plot5_tileseq.pdf&quot;,width=2.5,height=2.5,units = &quot;in&quot;,useDingbats=F)


ggplot()+
  geom_point(size=2,shape=21,fill=&quot;#3287BD&quot;,aes(x=.055-standards_ngs$drug_effect,y=standards_ngs$netgr_obs_screen1))+
  geom_errorbar(aes(x=.055-standards_ngs$drug_effect,ymin=standards_ngs$netgr_ll,ymax=standards_ngs$netgr_ul,width=0.001))+
  stat_cor(aes(x=.055-standards_ngs$drug_effect,y=standards_ngs$mean_fitness),method=&quot;pearson&quot;,color=&quot;#3287BD&quot;,size=2,vjust=3)+
scale_x_continuous(&quot;Predicted Net Growth Rate (Hrs-1) \nClonal IC50&quot;,limits=c(-0.016,.06))+
  scale_y_continuous(&quot;Observed Net Growth Rate (Hrs-1) \nDMS Screen&quot;,limits=c(-0.016,.06))+
  geom_abline(linetype=&quot;dashed&quot;)+
  cleanup+
  theme(legend.position = c(0.92,0.45),
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color=&quot;black&quot;,size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color=&quot;black&quot;,size=8),
        plot.margin=grid::unit(c(1,1,1,1), &quot;mm&quot;))</code></pre>
<p><img src="figure/ErrorRates_Tileseq_NGS_Duplex_SSCS.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;output/errorrates_figures/plot5_ngs.pdf&quot;,width=2.5,height=2.5,units = &quot;in&quot;,useDingbats=F)</code></pre>
</div>
<div id="archive" class="section level2">
<h2>Archive</h2>
<div id="prior-sscs-vs-tileseq-analyses." class="section level3">
<h3>Prior SSCS vs Tileseq analyses.</h3>
<pre class="r"><code>ggplot(data_snr_cast,aes(x=protein_start,y=snr))+geom_point()+facet_wrap(~seqtech)+scale_y_continuous(limits=c(0,60))

ggplot(data_snr_cast,aes(x=snr,fill=seqtech))+geom_density(alpha=0.6)+scale_x_continuous(&quot;Signal to noise&quot;,trans=&quot;log10&quot;)

ggplot(data_snr_cast,aes(x=protein_start,y=nmuts_intended))+geom_point()+facet_wrap(~seqtech)

ggplot(data_snr_cast,aes(x=protein_start,y=nmuts_unintended))+geom_point()+facet_wrap(~seqtech)

ggplot(data_snr_cast,aes(x=protein_start,y=coverage_mean_unintended))+geom_point()+facet_wrap(~seqtech)+scale_y_continuous(limits=c(0,200))

ggplot(data_snr_cast,aes(x=protein_start,y=coverage_sum_unintended))+geom_point()+facet_wrap(~seqtech)+scale_y_continuous(trans = &quot;log10&quot;)



ggplot(data%&gt;%filter(resmuts%in%T,is_intended%in%1),aes(x=species,y=ct_screen1_before,fill=seqtech))+geom_point(color=&quot;black&quot;,shape=21)+geom_line()+scale_y_continuous(&quot;Day 0 Coverage of Resistant Mutant&quot;,trans=&quot;log10&quot;)+cleanup+theme(axis.text.x=element_text(angle=90,hjust=.5,vjust=.5))

ggplot(data%&gt;%filter(resmuts%in%T,is_intended%in%1),aes(x=species,y=ct_screen1_after,fill=seqtech))+geom_point(color=&quot;black&quot;,shape=21)+geom_line()+scale_y_continuous(&quot;Day 6 Coverage of Resistant Mutant&quot;,trans=&quot;log10&quot;)+cleanup+theme(axis.text.x=element_text(angle=90,hjust=.5,vjust=.5))

ggplot(data%&gt;%filter(is_intended%in%1),aes(x=ct_screen1_before,color=seqtech))+stat_ecdf()+scale_x_continuous(&quot;Intended mutant coverage&quot;,trans=&quot;log10&quot;)+scale_y_continuous(&quot;CDF&quot;)


ggplot(data,aes(x=ct_screen1_before,fill=seqtech))+geom_histogram()+facet_wrap(seqtech~is_intended)+scale_x_continuous(trans=&quot;log10&quot;)


ggplot(data%&gt;%filter(is_intended%in%1),aes(x=ct_screen1_after,fill=seqtech))+geom_histogram()+facet_wrap(~seqtech)+scale_x_continuous(trans=&quot;log10&quot;)</code></pre>
<p>This is the analysis to allow benchmarking of ABL data to tileseq
data</p>
<p>This google doc has notes on how to interpret the google doc</p>
<pre class="r"><code># rm(list=ls())

data=read.csv(&quot;output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/data/TileSeqandSSCS_full_kinase_alldoses_with_lfc_corrected_netgrowths_12192024.csv&quot;,header = T,stringsAsFactors = F)
# a=data%&gt;%filter(is_intended%in%0,seqtech%in%&quot;sscs&quot;,dose%in%&quot;300nM&quot;,protein_start%in%c(242:512))
ggplot(data,aes(x=factor(protein_start),y=ct_screen1_before,fill=factor(is_intended)))+geom_boxplot()+facet_wrap(~seqtech,nrow = 2)+cleanup+scale_y_continuous(trans=&quot;log10&quot;)+theme(axis.text.x=element_text(angle=90, hjust=1))



data_snr=data%&gt;%filter(protein_start%in%c(242:512),!protein_start%in%c(354:361))%&gt;%group_by(seqtech,dose,is_intended,protein_start)%&gt;%summarize(nmuts=n(),coverage_mean=mean(ct_screen1_before),depth_mean=mean(depth_screen1_before))

data_snr_cast &lt;- data_snr %&gt;%
  pivot_wider(
    names_from = is_intended,
    values_from = c(nmuts,coverage_mean,depth_mean),
    names_glue = &quot;{.value}_{ifelse(is_intended == 1, &#39;intended&#39;, &#39;unintended&#39;)}&quot;
  )
data_snr_cast=data_snr_cast%&gt;%mutate(snr=coverage_mean_intended/coverage_mean_unintended)

ggplot(data_snr_cast%&gt;%filter(dose%in%&quot;300nM&quot;),aes(x=protein_start,y=snr))+geom_point()+facet_wrap(~seqtech)+scale_y_continuous(limits=c(0,60))

ggplot(data_snr_cast%&gt;%filter(dose%in%&quot;300nM&quot;),aes(x=snr,fill=seqtech))+geom_density(alpha=0.6)+scale_x_continuous(&quot;Signal to noise&quot;,trans=&quot;log10&quot;)

ggplot(data_snr_cast%&gt;%filter(dose%in%&quot;300nM&quot;),aes(x=protein_start,y=nmuts_intended))+geom_point()+facet_wrap(~seqtech)

ggplot(data_snr_cast%&gt;%filter(dose%in%&quot;300nM&quot;),aes(x=protein_start,y=nmuts_unintended))+geom_point()+facet_wrap(~seqtech)

ggplot(data_snr_cast%&gt;%filter(dose%in%&quot;300nM&quot;),aes(x=protein_start,y=coverage_mean_unintended))+geom_point()+facet_wrap(~seqtech)+scale_y_continuous(limits=c(0,200))

ggplot(data_snr_cast%&gt;%filter(dose%in%&quot;300nM&quot;),aes(x=coverage_mean_unintended/depth_mean_unintended,fill=seqtech))+geom_density(alpha=.6)+scale_x_continuous(&quot;Allele fraction of unintended mutants&quot;,trans=&quot;log10&quot;)

ggplot(data_snr_cast%&gt;%filter(dose%in%&quot;300nM&quot;),aes(x=coverage_mean_intended/depth_mean_intended,fill=seqtech))+geom_density(alpha=.6)+scale_x_continuous(&quot;Allele fraction of intended mutants&quot;,trans=&quot;log10&quot;)
# data_snr_cast=resmuts_adder(data_snr_cast)
# a=data%&gt;%filter(seqtech%in%&quot;sscs&quot;,dose%in%&quot;300nM&quot;,is_intended%in%1,protein_start%in%c(245))


ggplot(data%&gt;%filter(resmuts%in%T,dose%in%&quot;300nM&quot;,is_intended%in%1),aes(x=species,y=ct_screen1_before,fill=seqtech))+geom_point(color=&quot;black&quot;,shape=21)+geom_line()+scale_y_continuous(&quot;Day 0 Coverage of Resistant Mutant&quot;,trans=&quot;log10&quot;)+cleanup+theme(axis.text.x=element_text(angle=90,hjust=.5,vjust=.5))

ggplot(data%&gt;%filter(resmuts%in%T,dose%in%&quot;300nM&quot;,is_intended%in%1),aes(x=species,y=ct_screen1_after,fill=seqtech))+geom_point(color=&quot;black&quot;,shape=21)+geom_line()+scale_y_continuous(&quot;Day 6 Coverage of Resistant Mutant&quot;,trans=&quot;log10&quot;)+cleanup+theme(axis.text.x=element_text(angle=90,hjust=.5,vjust=.5))

ggplot(data%&gt;%filter(dose%in%&quot;300nM&quot;),aes(x=ct_screen1_before,))

ggplot(data%&gt;%filter(is_intended%in%1,dose%in%&quot;300nM&quot;),aes(x=ct_screen1_before,color=seqtech))+stat_ecdf()+scale_x_continuous(&quot;Intended mutant coverage&quot;,trans=&quot;log10&quot;)+scale_y_continuous(&quot;CDF&quot;)

ggplot(data%&gt;%filter(is_intended%in%1,dose%in%&quot;300nM&quot;),aes(x=ct_screen1_after,color=seqtech))+stat_ecdf()+scale_x_continuous(&quot;Intended mutant coverage&quot;,trans=&quot;log10&quot;)+scale_y_continuous(&quot;CDF&quot;)


ggplot(data%&gt;%filter(dose%in%&quot;300nM&quot;),aes(x=ct_screen1_before,fill=seqtech))+geom_histogram()+facet_wrap(seqtech~is_intended)+scale_x_continuous(trans=&quot;log10&quot;)

ggplot(data%&gt;%filter(is_intended%in%1,dose%in%&quot;300nM&quot;,seqtech==&quot;sscs&quot;),aes(x=ct_screen1_before))+geom_histogram()+scale_x_continuous(trans=&quot;log10&quot;)

ggplot(data%&gt;%filter(is_intended%in%1,dose%in%&quot;300nM&quot;,seqtech==&quot;sscs&quot;),aes(x=ct_screen1_after))+geom_histogram()+scale_x_continuous(trans=&quot;log10&quot;)

ggplot(data%&gt;%filter(dose%in%&quot;300nM&quot;,is_intended%in%1),aes(x=ct_screen1_after,fill=seqtech))+geom_histogram()+facet_wrap(~seqtech)+scale_x_continuous(trans=&quot;log10&quot;)


# Next steps: try to fit the poisson model and the negative binomial model to the count distributions. See if the confidence intervals of the negative binomial distributions are bigger for the SSCS data.
ggplot(data%&gt;%filter(is_intended%in%1,dose%in%&quot;300nM&quot;,seqtech==&quot;sscs&quot;),aes(x=ct_screen1_before))+geom_histogram()+scale_x_continuous(limits=c(0,500))

# a=data_snr_cast%&gt;%filter(dose%in%&quot;300nM&quot;)</code></pre>
<p>Tileseq data Fitting negative binomial and poissons to data</p>
<pre class="r"><code>sscsdata=data%&gt;%filter(is_intended%in%1,dose%in%&quot;300nM&quot;,seqtech==&quot;tileseq&quot;)
sscsdata=sscsdata%&gt;%filter(ct_screen1_before&lt;=10000)
sum(sscsdata$ct_screen1_before)
nb_fit=glm.nb(sscsdata$ct_screen1_before~1)
summary(nb_fit)

# Extract parameter estimates fir begatuve binomial
nb_mu &lt;- mean(sscsdata$ct_screen1_before)
nb_size &lt;- nb_fit$theta  # Dispersion parameter

# Fit a Poisson model (mean only)
poisson_mu &lt;- mean(sscsdata$ct_screen1_before)


# Generate a range of counts for plotting
count_range &lt;- 1:10000

# Calculate negative binomial probabilities for the range
nb_probs &lt;- dnbinom(count_range, size = nb_size, mu = nb_mu)

# Calculate probabilities for Poisson
poisson_probs &lt;- dpois(count_range, lambda = poisson_mu)

# plot(count_range, nb_probs, type = &quot;b&quot;, col = &quot;red&quot;, lwd = 2)

# Create data frames for the fits
fit_df &lt;- data.frame(
  count = rep(count_range, 2),
  density = c(nb_probs, poisson_probs),
  distribution = rep(c(&quot;Negative Binomial&quot;, &quot;Poisson&quot;), each = length(count_range))
)


ggplot(sscsdata)+
  geom_histogram(color=&quot;black&quot;,fill=&quot;gray90&quot;,aes(x=ct_screen1_before,y=..density..))+
  # geom_line(color=&quot;red&quot;,data=fit_df,aes(x=count,y=density))+
  # Negative binomial fit
  geom_line(
    data = fit_df[fit_df$distribution == &quot;Negative Binomial&quot;, ],
    aes(x = count, y = density, color = distribution), size = 1
  ) +
  # Poisson fit
  geom_line(
    data = fit_df[fit_df$distribution == &quot;Poisson&quot;, ], 
    aes(x = count, y = density, color = distribution), size = 1, linetype = &quot;dashed&quot;
  ) +
  # scale_y_continuous(trans=&quot;log10&quot;)+
  scale_x_continuous(&quot;Tileseq Mutant coverage&quot;,limits=c(1,10000))+
  scale_y_continuous(limits=c(0,.001))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] binom_1.1-1        tidyr_1.1.3        ggpubr_0.4.0       reshape2_1.4.4    
 [5] ggrepel_0.8.2      RColorBrewer_1.1-2 doParallel_1.0.15  iterators_1.0.12  
 [9] foreach_1.5.0      tictoc_1.0         plotly_4.9.2.1     ggplot2_3.3.3     
[13] dplyr_1.0.6        stringr_1.4.0     

loaded via a namespace (and not attached):
 [1] httr_1.4.2        sass_0.4.1        jsonlite_1.7.2    viridisLite_0.3.0
 [5] carData_3.0-3     bslib_0.3.1       assertthat_0.2.1  cellranger_1.1.0 
 [9] yaml_2.2.1        pillar_1.6.1      backports_1.1.7   glue_1.4.1       
[13] digest_0.6.25     promises_1.1.0    ggsignif_0.6.0    colorspace_1.4-1 
[17] htmltools_0.5.2   httpuv_1.5.2      plyr_1.8.6        pkgconfig_2.0.3  
[21] broom_0.7.6       haven_2.4.1       purrr_0.3.4       scales_1.1.1     
[25] openxlsx_4.1.5    later_1.0.0       rio_0.5.16        git2r_0.27.1     
[29] tibble_3.1.2      generics_0.0.2    farver_2.0.3      car_3.0-7        
[33] ellipsis_0.3.2    withr_2.4.2       lazyeval_0.2.2    magrittr_2.0.1   
[37] crayon_1.4.1      readxl_1.3.1      evaluate_0.14     fs_1.4.1         
[41] fansi_0.4.1       rstatix_0.6.0     forcats_0.5.1     foreign_0.8-78   
[45] tools_4.0.0       data.table_1.14.8 hms_1.1.0         lifecycle_1.0.0  
[49] munsell_0.5.0     zip_2.0.4         compiler_4.0.0    jquerylib_0.1.4  
[53] rlang_0.4.11      grid_4.0.0        htmlwidgets_1.5.1 labeling_0.3     
[57] rmarkdown_2.14    gtable_0.3.0      codetools_0.2-16  abind_1.4-5      
[61] DBI_1.1.0         curl_4.3          R6_2.4.1          knitr_1.28       
[65] fastmap_1.1.0     utf8_1.1.4        workflowr_1.6.2   rprojroot_1.3-2  
[69] stringi_1.7.5     Rcpp_1.0.4.6      vctrs_0.3.8       tidyselect_1.1.0 
[73] xfun_0.31        </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
