<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Haider Inam" />


<title>spikeins_e255k_gfp</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.9.2.1/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-1.52.2/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-1.52.2/plotly-latest.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">abl_dms</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">spikeins_e255k_gfp</h1>
<h4 class="author">Haider Inam</h4>
<h4 class="date">6/22/2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>abl_dms/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250621code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250621)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250621code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250621)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong253724b">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 253724b
</a>
</p>
</div>
<div id="strongRepositoryversionstrong253724b"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 253724b.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/Homepage_Spike_In_Mutants.Rmd
    Untracked:  analysis/dose_response_curve_fitting_with_errorbars.Rmd
    Untracked:  analysis/enrichment_simulations.Rmd
    Untracked:  analysis/spikeins_data_parser.Rmd
    Untracked:  analysis/spikeins_depthofcoverages.Rmd
    Untracked:  analysis/spikeins_e255k_gfp.Rmd
    Untracked:  analysis/spikeins_growthrates.Rmd
    Untracked:  data/Twinstrand/
    Untracked:  data/gfpenrichmentdata.csv
    Untracked:  data/ic50data/
    Untracked:  dosing_normalization_stdevs_paired.pdf
    Untracked:  output/Enrichmentsimulations/
    Untracked:  output/spikeins_figures/
    Untracked:  output/twinstrand_maf_merge.csv
    Untracked:  output/twinstrand_simple_melt_merge.csv

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code># rm(list=ls())
library(&quot;lmtest&quot;)</code></pre>
<pre><code>Loading required package: zoo</code></pre>
<pre><code>
Attaching package: &#39;zoo&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    as.Date, as.Date.numeric</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 4.0.2</code></pre>
<pre class="r"><code>library(&quot;MASS&quot;)</code></pre>
<pre><code>Warning: package &#39;MASS&#39; was built under R version 4.0.5</code></pre>
<pre class="r"><code>library(&quot;fitdistrplus&quot;)</code></pre>
<pre><code>Loading required package: survival</code></pre>
<pre><code>Loading required package: npsurv</code></pre>
<pre><code>Loading required package: lsei</code></pre>
<pre class="r"><code>library(&quot;lme4&quot;)</code></pre>
<pre><code>Loading required package: Matrix</code></pre>
<pre class="r"><code>library(&quot;boot&quot;)</code></pre>
<pre><code>
Attaching package: &#39;boot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:survival&#39;:

    aml</code></pre>
<pre class="r"><code>library(&quot;dplyr&quot;)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.0.2</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:MASS&#39;:

    select</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(&quot;plotly&quot;)</code></pre>
<pre><code>
Attaching package: &#39;plotly&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:MASS&#39;:

    select</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    last_plot</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre><code>The following object is masked from &#39;package:graphics&#39;:

    layout</code></pre>
<pre class="r"><code>library(&quot;reshape2&quot;)
library(RColorBrewer)
conc_for_predictions=0.8
net_gr_wodrug=0.05

# twinstrand_simple_melt_merge=read.csv(&quot;../output/twinstrand_simple_melt_merge.csv&quot;,header = T,stringsAsFactors = F)
twinstrand_simple_melt_merge=read.csv(&quot;data/twinstrand/twinstrand_simple_melt_merge.csv&quot;,header = T,stringsAsFactors = F)

# twinstrand_maf_merge=read.csv(&quot;../output/twinstrand_maf_merge.csv&quot;,header = T,stringsAsFactors = F)
twinstrand_maf_merge=read.csv(&quot;data/twinstrand/twinstrand_maf_merge.csv&quot;,header = T,stringsAsFactors = F)

# ic50data_long=read.csv(&quot;../output/ic50data_all_conc.csv&quot;,header = T,stringsAsFactors = F)
ic50data_long=read.csv(&quot;data/ic50data/ic50data_all_conc.csv&quot;,header = T,stringsAsFactors = F)
ic50data_long$netgr_pred=net_gr_wodrug-ic50data_long$drug_effect

# gfpdata=read.table(&quot;../data/gfpenrichmentdata.csv&quot;,stringsAsFactors = F,header = T,sep=&#39;,&#39;)
gfpdata=read.table(&quot;data/gfpenrichmentdata.csv&quot;,stringsAsFactors = F,header = T,sep=&#39;,&#39;)



# ic50_heatmap=read.csv(&quot;../data/IC50HeatMap.csv&quot;,header = T,stringsAsFactors = F)
ic50_heatmap=read.csv(&quot;data/ic50data/IC50HeatMap.csv&quot;,header = T,stringsAsFactors = F)

cleanup=theme_bw() +
  theme(plot.title = element_text(hjust=.5),
        panel.grid.major = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.background = element_blank(),
        # axis.line = element_line(color = &quot;black&quot;),
        axis.text = element_text(face=&quot;bold&quot;,color=&quot;black&quot;,size=&quot;11&quot;),
        text=element_text(size=11,face=&quot;bold&quot;),
        axis.title=element_text(face=&quot;bold&quot;,size=&quot;11&quot;))</code></pre>
<p>This boxplot will compare the net growth rates of E255K observed
across IC50s, FACs studies, and pooled studies</p>
<pre class="r"><code>###########Netgr from pooled approach###########
pooled_data=twinstrand_simple_melt_merge%&gt;%filter(mutant==&quot;E255K&quot;,!netgr_obs%in%NA,duration%in%c(&quot;d0d3&quot;,&quot;d0d6&quot;))%&gt;%dplyr::select(mutant,experiment,netgr=netgr_obs,Spike_in_freq)%&gt;%mutate(measurement=&quot;Pooled&quot;)
##Adding Wt growth rates
    #These decay rates were taken from wildtype_growthrates_sequenced.csv. These growthrates are generated on a per-residue basis by subtracting the sum of the altDepths from the Depth at that residue. For example, if residue T315 had a depth of 20,000 reads, and 5,000 of those reads were non-parental, then we would assume that there are 15,000 Wt reads. These calculations are also present in the &quot;nonlinear_growth_analysis.rmd&quot; code.
    # M3    -0.03618460529
    # M4    -0.02631366224
    # M5    -0.02841704794
    # M6    -0.04296255129
    # M7    -0.03614349619
pooled_data=pooled_data%&gt;%dplyr::select(mutant,measurement,netgr)
pooled_data=rbind(pooled_data,
        c(&quot;WT&quot;,&quot;Pooled&quot;,&quot;-0.03618460529&quot;),
        c(&quot;WT&quot;,&quot;Pooled&quot;,&quot;-0.02631366224&quot;),
        c(&quot;WT&quot;,&quot;Pooled&quot;,&quot;-0.02841704794&quot;),
        c(&quot;WT&quot;,&quot;Pooled&quot;,&quot;-0.04296255129&quot;),
        c(&quot;WT&quot;,&quot;Pooled&quot;,&quot;-0.03614349619&quot;))

#####The above E255K numbers are E255K sequenced in the experiments. The numbers below are pooled FACs numbers from the same experiments.
# Uncomment the lines below to include pooled_facs data in the plots below
# pooled_data=rbind(pooled_data,
        # c(&quot;E255K&quot;,&quot;Pooled_FACs&quot;,&quot;0.047882979&quot;),
        # c(&quot;E255K&quot;,&quot;Pooled_FACs&quot;,&quot;0.042670484&quot;),
        # c(&quot;E255K&quot;,&quot;Pooled_FACs&quot;,&quot;0.034028176&quot;),
        # c(&quot;E255K&quot;,&quot;Pooled_FACs&quot;,&quot;0.040478601&quot;),
        # c(&quot;E255K&quot;,&quot;Pooled_FACs&quot;,&quot;0.036180156&quot;))



pooled_data$netgr=as.numeric(pooled_data$netgr)
#Why does the GFP FACs data seem to have the same growth rates?

###########Netgr from IC50s###########
####E255K and WT#####
ic50_data=ic50_heatmap%&gt;%filter(species%in%c(&quot;E255K&quot;,&quot;WT&quot;))%&gt;%dplyr::select(mutant=species,y=X1.25)
ic50_data=ic50_data%&gt;%mutate(alpha=-log(y)/72)
ic50_data=ic50_data%&gt;%mutate(netgr=net_gr_wodrug-alpha,measurement=&quot;IC50s&quot;)
ic50_data=ic50_data%&gt;%dplyr::select(mutant,measurement,netgr)
###########Netgr from FACs experiment###########
#This data originally sits in the FACs entries in the &quot;Mixing Experiment 4, 8/20/18, directory in the Spike-ins for duplex sequencing project on benchling&quot;
#These are for both 1:1000 and 1:100 data. Growth rates looked similar at low frequencies with and without WT effects but there was a slowering of growth without WT spike-ins. We haven&#39;t focused on that here though
####E255k#####
facs_data=data.frame(c(0.046483696,0.0450497374,0.0197837978,0.0437170827,0.04520188,0.0440885505))
colnames(facs_data)=&quot;netgr&quot;
facs_data=facs_data%&gt;%mutate(mutant=&quot;E255K&quot;,Spike_in_freq=c(1000,1000,1000,10000,10000,10000),experiment=c(),measurement=&quot;FACS&quot;)

####WT#####
wt_facs_data=data.frame(c(-0.0152053893,-0.0159955607,-0.0330918372,-0.0230634613,-0.0262883706,-0.0288481251))
colnames(wt_facs_data)=&quot;netgr&quot;
wt_facs_data=wt_facs_data%&gt;%mutate(mutant=&quot;WT&quot;,Spike_in_freq=c(1000,1000,1000,10000,10000,10000),experiment=c(),measurement=&quot;FACS&quot;)

facs_data=rbind(facs_data,wt_facs_data)
facs_data=facs_data%&gt;%dplyr::select(mutant,measurement,netgr)


###########Plotting###########
###Please note that 625nM imatinib was used in all conditions
e255k_gr=rbind(facs_data,pooled_data,ic50_data)
e255k_gr$measurement=factor(e255k_gr$measurement,levels=c(&quot;IC50s&quot;,&quot;FACS&quot;,&quot;Pooled&quot;))

###Showing net growth rate###

ggplot(e255k_gr,aes(x=factor(measurement),y=netgr,fill=measurement))+
  geom_boxplot()+
  facet_wrap(~mutant)+
  scale_fill_manual(values = c(&quot;#63A088&quot;,&quot;#1CCE16&quot;,&quot;#206A36&quot;))+
  scale_y_continuous(name=&quot;Net Grwoth Rate Observed&quot;)+
  cleanup+
  theme(legend.position = &quot;none&quot;,
        axis.text.x=element_text(angle=15,hjust=.5,vjust=.5),
        axis.title.x=element_blank())</code></pre>
<p><img src="figure/spikeins_e255k_gfp.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(e255k_gr,aes(x=factor(measurement),y=netgr,fill=mutant))+
  geom_boxplot()+
  facet_wrap(~mutant)+
  scale_fill_manual(values = c(&quot;#1CCE16&quot;,&quot;gray90&quot;))+
  scale_y_continuous(name=&quot;Net growth rate (Hrs^-1)&quot;)+
  cleanup+
  theme(legend.position = &quot;none&quot;,
        axis.text.x=element_text(angle=15,hjust=.5,vjust=.5),
        axis.title.x=element_blank(),
        strip.text = element_blank(),
        axis.text.x.bottom=element_blank())</code></pre>
<p><img src="figure/spikeins_e255k_gfp.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&quot;output/spikeins_figures/e255k_wt_alphas_figure.pdf&quot;,width = 4,height = 2.5,units=&quot;in&quot;,useDingbats=F)</code></pre>
<pre class="r"><code>gfpdata$ttotal_sequenced=c(0,3,6)
gfpdata$xtotal_sequenced=c(129,3323,37023)
# gfpdata_simple=gfpdata%&gt;%dplyr::select(ttotal_4,xtotal_4_e255k)
# e255k=twinstrand_maf_merge%&gt;%filter(mutant==&quot;E255K&quot;,experiment==&quot;M3&quot;)

ggplot(gfpdata)+
  geom_line(aes(x=t_out_4,y=x_out_4_e255k),color=&quot;#1cce16&quot;,size=1.5)+
  geom_ribbon(aes(x =t_out_4_conintub,ymax=x_out_4_e255k_ciub,ymin=x_out_4_e255k_cilb),fill=&quot;#1cce16&quot;,alpha=.3)+
  # geom_line(aes(x=t_out_3,y=log10(x_out_3_e255k)),color=&quot;#206A36&quot;,size=1.5)+
  # geom_ribbon(aes(x =t_out_3_conintub,ymax=log10(x_out_3_e255k_ciub),ymin=log10(x_out_3_e255k_cilb)),fill=&quot;#206A36&quot;,alpha=.3)+
  geom_point(aes(x=ttotal_4,y=xtotal_4_e255k),size=3)+
  # geom_point(aes(x=ttotal_3,y=xtotal_3_e255k),size=3)+
  geom_point(color=&quot;black&quot;,shape=21,size=3,aes(x=ttotal_sequenced,y=xtotal_sequenced,fill=&quot;orange&quot;))+
  scale_x_continuous(name=&#39;Time (Days)&#39;,limits=c(0,7))+
  scale_y_continuous(trans=&quot;log10&quot;,name=&#39;Resistant Population&#39;,labels=parse(text = c(&quot;10^3&quot;,&quot;10^5&quot;)),breaks = c(10^3,10^5),limits=c(10^2,10^6))+
  theme_bw()+
  theme(plot.title = element_text(hjust=.5),
        legend.position=&quot;none&quot;,
        text = element_text(size=10),
        axis.text=element_text(face=&quot;bold&quot;,size=&quot;10&quot;,color=&quot;black&quot;),
        axis.text.y =element_text(face=&quot;bold&quot;,size=&quot;10&quot;,color=&quot;black&quot;),
        axis.title=element_text(face=&quot;bold&quot;,size=&quot;10&quot;,color=&quot;black&quot;))</code></pre>
<pre><code>Warning: Removed 13 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning: Removed 40 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/spikeins_e255k_gfp.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # scale_y_continuous(labels = function(x) format(x, scientific = TRUE), limits = c(1e2,5e6),trans=&#39;log10&#39;,labels=parse(text = c(&quot;10^3&quot;,&quot;10^5&quot;))+
  # scale_y_continuous(labels = parse(text = c(&quot;10^3&quot;,&quot;10^5&quot;)), limits = c(1e2,5e6),trans=&#39;log10&#39;)+

# ggsave(&quot;e255k_initial_spikins_figure.pdf&quot;,width=3,height=3,units=&quot;in&quot;,useDingbats=F)</code></pre>
<p>Looking at differences in growth rate between different
replicates</p>
<pre class="r"><code>#Fetching E255K growth rates that we saw in sequencing
e255k_sequenced=twinstrand_simple_melt_merge%&gt;%filter(mutant%in%&quot;E255K&quot;,duration%in%&quot;d3d6&quot;)
e255k_sequenced_2=e255k_sequenced%&gt;%mutate(measurement=&quot;Sequenced&quot;)%&gt;%dplyr::select(Experiment=experiment,measurement,netgr_obs)
#Adding E255K growth rates that we saw during flow cytometry. These are the mean net growth rates from D3 to D6. Did not do D0 to D6 because we did not sequence D0 for all replicates and hence it would be an unfair comparison.
e255k_sequenced_facs=rbind(e255k_sequenced_2,
                 c(&quot;M3&quot;,&quot;FACs&quot;,0.035096215),
                 c(&quot;M4&quot;,&quot;FACs&quot;,0.028365649),
                 c(&quot;M5&quot;,&quot;FACs&quot;,0.016017951),
                 c(&quot;M6&quot;,&quot;FACs&quot;,0.02481819),
                 c(&quot;M7&quot;,&quot;FACs&quot;,0.026915217))

e255k_sequenced_facs$netgr_obs=as.numeric(as.character(e255k_sequenced_facs$netgr_obs))

#In the next step, I&#39;m ordering mutants by decreasing resposne to the 625nM dose. Then I use this to change the levels of the species factor from more to less resistant. This helps with ggplot because now I can color the mutants with decreasing resistance
e255k_sequenced_facs$Experiment=factor(e255k_sequenced_facs$Experiment,levels = as.character(e255k_sequenced_facs$Experiment[order((e255k_sequenced_2$netgr_obs),decreasing = T)]))

ggplot(e255k_sequenced_facs,aes(x=Experiment,y=netgr_obs,fill=measurement))+
  geom_col(position=&quot;dodge&quot;)+
  scale_y_continuous(limits = c(0,NA))+
  scale_fill_manual(values=c(&quot;#1CCE16&quot;,&quot;#206A36&quot;))+
  cleanup</code></pre>
<p><img src="figure/spikeins_e255k_gfp.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>getPalette = colorRampPalette(brewer.pal(6, &quot;Spectral&quot;))
ggplot(e255k_sequenced_facs,aes(x=measurement,y=netgr_obs,group=Experiment))+
  geom_line()+
  geom_point(color=&quot;black&quot;,size=3,shape=21,aes(fill=Experiment))+
  cleanup+
  scale_y_continuous(name=&quot;Observed growth rate&quot;,limits = c(0,NA))+
  scale_fill_manual(values=getPalette(length(unique(e255k_sequenced_facs$Experiment))))+
  theme(axis.title.x = element_blank())</code></pre>
<p><img src="figure/spikeins_e255k_gfp.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(axis.title.x = element_blank(),
        # legend.position = c(.9,.2))

# ggsave(&quot;e255k_facs_sequenced.pdf&quot;,width=4,height = 3,units = &quot;in&quot;,useDingbats=F)</code></pre>
<p>Looking at the growth rate changes with dates</p>
<pre class="r"><code>a=twinstrand_simple_melt_merge%&gt;%filter(duration%in%c(&quot;d0d3&quot;,&quot;d3d6&quot;),experiment%in%c(&quot;M3&quot;,&quot;M6&quot;))
plotly=ggplot(a,aes(x=factor(mutant),y=netgr_obs,fill=duration))+geom_col(position=&quot;dodge&quot;,stat=&quot;identity&quot;)+facet_wrap(~experiment)</code></pre>
<pre><code>Warning: Ignoring unknown parameters: stat</code></pre>
<pre class="r"><code>ggplotly(plotly)</code></pre>
<pre><code>Warning: `group_by_()` was deprecated in dplyr 0.7.0.
Please use `group_by()` instead.
See vignette(&#39;programming&#39;) for more help</code></pre>
<div id="htmlwidget-0450937840dd11870dbd" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-0450937840dd11870dbd">{"x":{"data":[{"orientation":"v","width":[0.450000000000003,0.450000000000003,0.45,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.450000000000003,0.449999999999999,0.45,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.45,0.449999999999999,0.450000000000003,0.45],"base":[0,0,0,0,0,0,0,0,0,0,0,0,-0.0031374685958978,0,0,0,0,0,null,0],"x":[16.775,18.775,2.775,14.775,10.775,11.775,13.775,12.775,17.775,15.775,1.775,6.775,8.775,9.775,7.775,4.775,3.775,5.775,19.775,0.775],"y":[0.0616556918320131,0.0691322117935927,0.0626300306066603,0.0509812388688562,0.058362136592558,0.0513796625167495,0.0458359814430316,0.0374419279441509,0.032550335908946,0.0353917597999288,0.0450944276782183,0.0315381611218042,0.0031374685958978,0.034874909856677,0.0229487230196449,0.0185280300274317,0.0349377398118437,0.0232793848963411,null,0.0177152567264708],"text":["factor(mutant): T315I<br />netgr_obs:  0.0616556918<br />duration: d0d3","factor(mutant): Y253H<br />netgr_obs:  0.0691322118<br />duration: d0d3","factor(mutant): E255V<br />netgr_obs:  0.0626300306<br />duration: d0d3","factor(mutant): M351T<br />netgr_obs:  0.0509812389<br />duration: d0d3","factor(mutant): G250E<br />netgr_obs:  0.0583621366<br />duration: d0d3","factor(mutant): H396P<br />netgr_obs:  0.0513796625<br />duration: d0d3","factor(mutant): L248V<br />netgr_obs:  0.0458359814<br />duration: d0d3","factor(mutant): H396R<br />netgr_obs:  0.0374419279<br />duration: d0d3","factor(mutant): Y253F<br />netgr_obs:  0.0325503359<br />duration: d0d3","factor(mutant): Q252H<br />netgr_obs:  0.0353917598<br />duration: d0d3","factor(mutant): E255K<br />netgr_obs:  0.0450944277<br />duration: d0d3","factor(mutant): F317L<br />netgr_obs:  0.0315381611<br />duration: d0d3","factor(mutant): F359L<br />netgr_obs: -0.0031374686<br />duration: d0d3","factor(mutant): F359V<br />netgr_obs:  0.0348749099<br />duration: d0d3","factor(mutant): F359I<br />netgr_obs:  0.0229487230<br />duration: d0d3","factor(mutant): E355G<br />netgr_obs:  0.0185280300<br />duration: d0d3","factor(mutant): E355A<br />netgr_obs:  0.0349377398<br />duration: d0d3","factor(mutant): E459K<br />netgr_obs:  0.0232793849<br />duration: d0d3","factor(mutant): Y320C<br />netgr_obs:            NA<br />duration: d0d3","factor(mutant): D276G<br />netgr_obs:  0.0177152567<br />duration: d0d3"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"d0d3","legendgroup":"d0d3","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.450000000000003,0.450000000000003,0.45,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.450000000000003,0.449999999999999,0.45,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.45,0.449999999999999,0.450000000000003],"base":[0,0,0,0,0,0,0,0,0,0,0,0,null,null,0,0,0,0,null],"x":[16.775,18.775,2.775,14.775,10.775,11.775,13.775,12.775,17.775,15.775,1.775,6.775,8.775,9.775,7.775,4.775,3.775,5.775,19.775],"y":[0.0644298708294598,0.070689834124514,0.0813387855024452,0.0711151434288902,0.0781230706856515,0.0671943722893807,0.0642524140765957,0.0398941155880398,0.0611217197241139,0.0380807945834078,0.0452084170790659,0.0384925557400118,null,null,0.0349286087440826,0.0138757912325283,0.037364718698236,0.0430977566389742,null],"text":["factor(mutant): T315I<br />netgr_obs:  0.0644298708<br />duration: d0d3","factor(mutant): Y253H<br />netgr_obs:  0.0706898341<br />duration: d0d3","factor(mutant): E255V<br />netgr_obs:  0.0813387855<br />duration: d0d3","factor(mutant): M351T<br />netgr_obs:  0.0711151434<br />duration: d0d3","factor(mutant): G250E<br />netgr_obs:  0.0781230707<br />duration: d0d3","factor(mutant): H396P<br />netgr_obs:  0.0671943723<br />duration: d0d3","factor(mutant): L248V<br />netgr_obs:  0.0642524141<br />duration: d0d3","factor(mutant): H396R<br />netgr_obs:  0.0398941156<br />duration: d0d3","factor(mutant): Y253F<br />netgr_obs:  0.0611217197<br />duration: d0d3","factor(mutant): Q252H<br />netgr_obs:  0.0380807946<br />duration: d0d3","factor(mutant): E255K<br />netgr_obs:  0.0452084171<br />duration: d0d3","factor(mutant): F317L<br />netgr_obs:  0.0384925557<br />duration: d0d3","factor(mutant): F359L<br />netgr_obs:            NA<br />duration: d0d3","factor(mutant): F359V<br />netgr_obs:            NA<br />duration: d0d3","factor(mutant): F359I<br />netgr_obs:  0.0349286087<br />duration: d0d3","factor(mutant): E355G<br />netgr_obs:  0.0138757912<br />duration: d0d3","factor(mutant): E355A<br />netgr_obs:  0.0373647187<br />duration: d0d3","factor(mutant): E459K<br />netgr_obs:  0.0430977566<br />duration: d0d3","factor(mutant): Y320C<br />netgr_obs:            NA<br />duration: d0d3"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"d0d3","legendgroup":"d0d3","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.450000000000003,0.450000000000003,0.45,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.450000000000003,0.450000000000003,0.45,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.45,0.449999999999999,0.450000000000003,0.45],"base":[0,0,0,0,0,0,0,0,0,0,0,0,null,0,0,0,0,0,0,null],"x":[17.225,19.225,3.225,15.225,11.225,12.225,14.225,13.225,18.225,16.225,2.225,7.225,9.225,10.225,8.225,5.225,4.225,6.225,20.225,1.225],"y":[0.0556532112719625,0.0566953578581641,0.0572690119933471,0.0550596784448217,0.0517196077013197,0.0487243231157813,0.0396406266927503,0.0427198643829193,0.0469383450557703,0.0350310392612771,0.0334809738680772,0.0308041892913613,null,0.028451268722294,0.0276739573164411,0.0267902770166541,0.012731599110772,0.00792197717242642,0.0441715968666497,null],"text":["factor(mutant): T315I<br />netgr_obs:  0.0556532113<br />duration: d3d6","factor(mutant): Y253H<br />netgr_obs:  0.0566953579<br />duration: d3d6","factor(mutant): E255V<br />netgr_obs:  0.0572690120<br />duration: d3d6","factor(mutant): M351T<br />netgr_obs:  0.0550596784<br />duration: d3d6","factor(mutant): G250E<br />netgr_obs:  0.0517196077<br />duration: d3d6","factor(mutant): H396P<br />netgr_obs:  0.0487243231<br />duration: d3d6","factor(mutant): L248V<br />netgr_obs:  0.0396406267<br />duration: d3d6","factor(mutant): H396R<br />netgr_obs:  0.0427198644<br />duration: d3d6","factor(mutant): Y253F<br />netgr_obs:  0.0469383451<br />duration: d3d6","factor(mutant): Q252H<br />netgr_obs:  0.0350310393<br />duration: d3d6","factor(mutant): E255K<br />netgr_obs:  0.0334809739<br />duration: d3d6","factor(mutant): F317L<br />netgr_obs:  0.0308041893<br />duration: d3d6","factor(mutant): F359L<br />netgr_obs:            NA<br />duration: d3d6","factor(mutant): F359V<br />netgr_obs:  0.0284512687<br />duration: d3d6","factor(mutant): F359I<br />netgr_obs:  0.0276739573<br />duration: d3d6","factor(mutant): E355G<br />netgr_obs:  0.0267902770<br />duration: d3d6","factor(mutant): E355A<br />netgr_obs:  0.0127315991<br />duration: d3d6","factor(mutant): E459K<br />netgr_obs:  0.0079219772<br />duration: d3d6","factor(mutant): Y320C<br />netgr_obs:  0.0441715969<br />duration: d3d6","factor(mutant): D276G<br />netgr_obs:            NA<br />duration: d3d6"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"d3d6","legendgroup":"d3d6","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.450000000000003,0.450000000000003,0.45,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.450000000000003,0.450000000000003,0.45,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.449999999999999,0.45,0.449999999999999,0.450000000000003],"base":[0,0,0,0,0,0,0,0,0,0,0,0,null,0,0,0,null,-0.00721304443411948,0],"x":[17.225,19.225,3.225,15.225,11.225,12.225,14.225,13.225,18.225,16.225,2.225,7.225,9.225,10.225,8.225,5.225,4.225,6.225,20.225],"y":[0.0564405227297085,0.0577807399467394,0.0565968051157205,0.0504638392474484,0.0511485233285775,0.0407904899805769,0.0302288352512854,0.0417233554390856,0.0382051517146049,0.0222071460834166,0.0223612985762536,0.0160500200244016,null,0.0297729630991121,0.000317413310529148,0.0118278860192327,null,0.00721304443411948,0.0294304233077991],"text":["factor(mutant): T315I<br />netgr_obs:  0.0564405227<br />duration: d3d6","factor(mutant): Y253H<br />netgr_obs:  0.0577807399<br />duration: d3d6","factor(mutant): E255V<br />netgr_obs:  0.0565968051<br />duration: d3d6","factor(mutant): M351T<br />netgr_obs:  0.0504638392<br />duration: d3d6","factor(mutant): G250E<br />netgr_obs:  0.0511485233<br />duration: d3d6","factor(mutant): H396P<br />netgr_obs:  0.0407904900<br />duration: d3d6","factor(mutant): L248V<br />netgr_obs:  0.0302288353<br />duration: d3d6","factor(mutant): H396R<br />netgr_obs:  0.0417233554<br />duration: d3d6","factor(mutant): Y253F<br />netgr_obs:  0.0382051517<br />duration: d3d6","factor(mutant): Q252H<br />netgr_obs:  0.0222071461<br />duration: d3d6","factor(mutant): E255K<br />netgr_obs:  0.0223612986<br />duration: d3d6","factor(mutant): F317L<br />netgr_obs:  0.0160500200<br />duration: d3d6","factor(mutant): F359L<br />netgr_obs:            NA<br />duration: d3d6","factor(mutant): F359V<br />netgr_obs:  0.0297729631<br />duration: d3d6","factor(mutant): F359I<br />netgr_obs:  0.0003174133<br />duration: d3d6","factor(mutant): E355G<br />netgr_obs:  0.0118278860<br />duration: d3d6","factor(mutant): E355A<br />netgr_obs:            NA<br />duration: d3d6","factor(mutant): E459K<br />netgr_obs: -0.0072130444<br />duration: d3d6","factor(mutant): Y320C<br />netgr_obs:  0.0294304233<br />duration: d3d6"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"d3d6","legendgroup":"d3d6","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":37.9178082191781,"r":7.30593607305936,"b":40.1826484018265,"l":48.9497716894977},"plot_bgcolor":"rgba(235,235,235,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,0.489128071319852],"automargin":true,"type":"linear","autorange":false,"range":[0.4,20.6],"tickmode":"array","ticktext":["D276G","E255K","E255V","E355A","E355G","E459K","F317L","F359I","F359L","F359V","G250E","H396P","H396R","L248V","M351T","Q252H","T315I","Y253F","Y253H","Y320C"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20],"categoryorder":"array","categoryarray":["D276G","E255K","E255V","E355A","E355G","E459K","F317L","F359I","F359L","F359V","G250E","H396P","H396R","L248V","M351T","Q252H","T315I","Y253F","Y253H","Y320C"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"annotations":[{"text":"factor(mutant)","x":0.5,"y":-0.0471841704718417,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"top","annotationType":"axis"},{"text":"netgr_obs","x":-0.0510980647966949,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"M3","x":0.244564035659926,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.689497716895},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"M6","x":0.755435964340074,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.689497716895},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"duration","x":1.02,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"left","yanchor":"bottom","legendTitle":true}],"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-0.0116406359309477,0.0857663769992734],"tickmode":"array","ticktext":["0.00","0.02","0.04","0.06","0.08"],"tickvals":[0,0.02,0.04,0.06,0.08],"categoryorder":"array","categoryarray":["0.00","0.02","0.04","0.06","0.08"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.489128071319852,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(217,217,217,1)","line":{"color":"transparent","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":0.489128071319852,"y0":0,"y1":23.37899543379,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.510871928680148,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(217,217,217,1)","line":{"color":"transparent","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.510871928680148,"x1":1,"y0":0,"y1":23.37899543379,"yanchor":1,"ysizemode":"pixel"}],"xaxis2":{"type":"linear","autorange":false,"range":[0.4,20.6],"tickmode":"array","ticktext":["D276G","E255K","E255V","E355A","E355G","E459K","F317L","F359I","F359L","F359V","G250E","H396P","H396R","L248V","M351T","Q252H","T315I","Y253F","Y253H","Y320C"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20],"categoryorder":"array","categoryarray":["D276G","E255K","E255V","E355A","E355G","E459K","F317L","F359I","F359L","F359V","G250E","H396P","H396R","L248V","M351T","Q252H","T315I","Y253F","Y253H","Y320C"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"domain":[0.510871928680148,1],"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"y":0.913385826771654},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"68c7f3b7f3":{"x":{},"y":{},"fill":{},"type":"bar"}},"cur_data":"68c7f3b7f3","visdat":{"68c7f3b7f3":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>#Realizing that the signal I&#39;m seeing here might be confounded by erroneous flow cytometry counts... </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RColorBrewer_1.1-2  reshape2_1.4.4      plotly_4.9.2.1     
 [4] dplyr_1.0.6         boot_1.3-24         lme4_1.1-23        
 [7] Matrix_1.2-18       fitdistrplus_1.0-14 npsurv_0.4-0.1     
[10] lsei_1.2-0.1        survival_3.1-12     MASS_7.3-55        
[13] ggplot2_3.3.3       lmtest_0.9-37       zoo_1.8-8          

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6      lattice_0.20-41   tidyr_1.1.3       assertthat_0.2.1 
 [5] rprojroot_1.3-2   digest_0.6.25     utf8_1.1.4        plyr_1.8.6       
 [9] R6_2.4.1          backports_1.1.7   evaluate_0.14     httr_1.4.2       
[13] pillar_1.6.1      rlang_0.4.11      lazyeval_0.2.2    minqa_1.2.4      
[17] data.table_1.14.8 jquerylib_0.1.4   nloptr_1.2.2.1    rmarkdown_2.14   
[21] labeling_0.3      splines_4.0.0     statmod_1.4.34    stringr_1.4.0    
[25] htmlwidgets_1.5.1 munsell_0.5.0     compiler_4.0.0    httpuv_1.5.2     
[29] xfun_0.31         pkgconfig_2.0.3   htmltools_0.5.2   tidyselect_1.1.0 
[33] tibble_3.1.2      workflowr_1.6.2   fansi_0.4.1       viridisLite_0.3.0
[37] crayon_1.4.1      withr_2.4.2       later_1.0.0       grid_4.0.0       
[41] nlme_3.1-147      jsonlite_1.7.2    gtable_0.3.0      lifecycle_1.0.0  
[45] DBI_1.1.0         git2r_0.27.1      magrittr_2.0.1    scales_1.1.1     
[49] stringi_1.7.5     farver_2.0.3      fs_1.4.1          promises_1.1.0   
[53] bslib_0.3.1       ellipsis_0.3.2    generics_0.0.2    vctrs_0.3.8      
[57] tools_4.0.0       glue_1.4.1        purrr_0.3.4       crosstalk_1.1.0.1
[61] fastmap_1.1.0     yaml_2.2.1        colorspace_1.4-1  knitr_1.28       
[65] sass_0.4.1       </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
